docker/docker.go

```diff
@@ -267,6 +289,131 @@ func (i *Image) Pull() error {
 	return nil
 }
 
+func (i *Image) GetFsCommands() []*FsLayerCommand {
+	return i.FsCommands
+}
+
+// FetchFsCommands retrieves information about image layers commands from docker registry.
+func (i *Image) FetchFsCommands() error {
+	if i.FsCommands != nil {
+		log.Infof("Layer commands are already present")
+		return nil
+	}
+	ctx := context.Background()
+
+	opts := fanal_types.DockerOption{
+		Timeout:  90*time.Second,
+		SkipPing: true,
+		UserName: i.user,
+		Password: i.password,
+	}
+	img, cleanup, err := newDockerImage(ctx, i.imageName, opts)
+	if err != nil {
+		return fmt.Errorf("failed to get docker image: %v", err)
+	}
+	defer cleanup()
+
+	conf, err := img.ConfigFile()
+	if err != nil {
+		return fmt.Errorf("failed to get config file: %v", err)
+	}
+	confB, err := json.Marshal(conf)
+	if err != nil {
+		return fmt.Errorf("failed to marshal config: %v", err)
+	}
+	log.Debugf("Image config: %s", confB)
+
+	var commands []string
+	for i, layerHistory := range conf.History {
+		if layerHistory.EmptyLayer {
+			log.Infof("Skipping empty layer (%v): %+v", i, layerHistory)
```

Comment:
Didn't we wanted to also show empty layers?

Reply:
I think that not currently from talking with Rafi. We can add this as part of layer support in dockle